Add a simple way of starting the monitoring stack using docker-compose (#1181)
scylladb/scylla-monitoring: docs/source/docker_compose.rst

Using Docker Compose
====================

Scylla-Monitor is container-based, using Docker Compose file allows you to simplify starting and stopping the monitoring stack.

> 

Reviewer: ```suggestion
.. warning:: 

    docker-compose and start_all.sh are two **alternatives** ways to launch Scylla Monitoring Stack. You should use one or the other, not both. In particular,  creating and updating *docker-compose.yml* is ignored when using *start_all.sh*

```

Reviewer 2: ```suggestion
.. warning:: 
    *docker-compose* **and** *start_all.sh* are two **alternative** ways to launch Scylla Monitoring Stack. You should use **one** method, **not both**. In particular,  creating and updating *docker-compose.yml* is ignored when using *start_all.sh*
```

Prerequisite 
------------

Make sure you have `docker` and `docker-compose` installed.

Setting Prometheus
------------------

> Prometheus configuration file contains among others the IP address of the alertmanager and either the location

Reviewer 2: ```suggestion
The Prometheus configuration file contains among others the IP address of the *alertmanager* and either the location
```

> of the `scylla_server.yml` file or a Consul IP address of Scylla-Manager, if Scylla-Manager is used for server provisioning.

Reviewer 2: ```suggestion
of the *scylla_server.yml* file or a Consul IP address of Scylla Manager, if Scylla Manager is used for server provisioning.
```

You can use `./prometheus-config.sh` to generate the file, for example: 

.. code-block:: shell

   ./prometheus-config.sh --compose
   

> For production usage, It is advice that you will use an external directory for the Prometheus database, make sure to create one and

Reviewer: ```suggestion
For production usage, It is adviced that you will use an external directory for the Prometheus database, make sure to create one and
```

> For production usage, It is advice that you will use an external directory for the Prometheus database, make sure to create one and

Reviewer 2: ```suggestion
For production systems, It is advised to use an external directory for the Prometheus database. Make sure to create one and update the docker-compose file accordingly (see the docker-compose example).  

```

Reviewer 2: add a link to the example

Author: The example is part of this document

Reviewer 2: so add a :ref: to that line.

> update the docker-compose file accordingly (see the docker-compose example).  

Reviewer 2: ```suggestion
```

Setting Grafana Provisioning
----------------------------

> Grafana reads its provisioning configuration from files, one for the data-source and one

Reviewer 2: from which files

Reviewer 2: If the files are explained below, we should link to the below sections. If this is some other file, we should either explain the location or link to where it is explained.

Author: It reads the configuration from files, i.e. the method of configuration is file

> Grafana reads its provisioning configuration from files, one for the data-source and one
> for the dashboards. Note that the latter tells Grafana where the dashboards are not the dashboard themselves that are
> in a different location.

Reviewer: ```suggestion
Grafana reads its provisioning configuration from files, one for the data-source and one
for the dashboards. Note that the latter tells Grafana where the dashboards are not the dashboard themselves in a different location.
```

Grafana Data-Source file
^^^^^^^^^^^^^^^^^^^^^^^^

> to update the datasource you can run

Reviewer 2: ```suggestion
Run the following command to update the datasource:
```

.. code-block:: shell

   ./grafana-datasource.sh --compose

> You can see the generated file under: `grafana/provisioning/datasources/datasource.yaml` 

Reviewer 2: ```suggestion
This command generates the following file: `grafana/provisioning/datasources/datasource.yaml` 
```

Grafana Dashboard Load file
^^^^^^^^^^^^^^^^^^^^^^^^^^^

> To set the dashboard load file, you can use the `./generate-sashboards.sh` with the `-t` command line flag, for example

Reviewer 2: ```suggestion
To set the dashboard load file, you can run the `./generate-dashboards.sh` script with the `-t` command line flag and the -v flag to specify the version.
```

> to use scylla-enterprise version 2020.1

Reviewer 2: ```suggestion
For example, Scylla Enterprise version 2020.1:
```

.. code-block:: shell

   ./generate-dashboards.sh -t -v 2020.1

> You can see the generated files `grafana/provisioning/dashboards/`

Reviewer 2: ```suggestion
This command generates the following files: `grafana/provisioning/dashboards/`
```

Docker Compose file
-------------------
You can use the following example as a base for your docker compose.

Pass the following to a file called `docker-compose.yml`

> .. code-block:: yaml

Reviewer: keep as a yaml file in git and include it

        - ./loki/rules:/etc/loki/rules
        - ./loki/conf:/mnt/config
      promotheus:
        command:
        - --config.file=/etc/prometheus/prometheus.compose.yml
        container_name: aprom
        image: prom/prometheus:v2.18.1
        ports:
        - 9090:9090
        volumes:
        - ./prometheus/prometheus.yml:/etc/prometheus/prometheus.yml
        - ./prometheus/prometheus.rules.yml:/etc/prometheus/prometheus.rules.yml
        - ./prometheus/scylla_servers.yml:/etc/scylla.d/prometheus/scylla_servers.yml
        - ./prometheus/scylla_manager_servers.yml:/etc/scylla.d/prometheus/scylla_manager_servers.yml
        - ./prometheus/scylla_servers.yml:/etc/scylla.d/prometheus/node_exporter_servers.yml
        # Uncomment the following line for prometheus persistency 
        # - path/to/data/dir:/prometheus/data
      promtail:
        command:
        - --config.file=/etc/promtail/config.yml
        container_name: promtail
        image: grafana/promtail:2.0.0
        ports:
        - 1514:1514
        - 9080:9080
        volumes:
        - ./loki/promtail/promtail_config.compose.yml:/etc/promtail/config.yml
    version: '3'

Start and Stop
^^^^^^^^^^^^^^

> use `docker-compose up` and `docker-compose down` to start and stop the monitoring stack. 

Reviewer 2: ```suggestion
To start the Scylla Monitoring Stack run ``docker-compose up`` and to stop run ``docker-compose down``. 
```
